src/application.vala

```diff
@@ -232,24 +224,109 @@ public class Application : Gtk.Application {
         }
     }
 
-    private void show_qrcode() {
+    private void create_cert() {
+        GnuTLS.X509.PrivateKey key = GnuTLS.X509.PrivateKey.create();
+        key.generate(GnuTLS.PKAlgorithm.RSA, 2048);
+        //GnuTLS.X509.Certificate certi = GnuTLS.X509.Certificate.create();
+    }
+
+    private string? call_sytem_function (string[] args) {
         try {
-            string read;
-            FileUtils.get_contents ("server.pem", out read);
+            string[] spawn_args = args;
+            string[] spawn_env = Environ.get ();
+            string ls_stdout;
+            string ls_stderr;
+            int ls_status;
+
+            Process.spawn_sync (".",
+                                spawn_args,
+                                spawn_env,
+                                SpawnFlags.SEARCH_PATH,
+                                null,
+                                out ls_stdout,
+                                out ls_stderr,
+                                out ls_status);
+            /*
+            stdout.printf ("stdout:\n");
+            stdout.puts (ls_stdout);
+            stdout.printf ("stderr:\n");
+            stdout.puts (ls_stderr);
+            stdout.printf ("status: %d\n", ls_status);*/ 
+            return ls_stdout;
+
+        } catch (SpawnError e) {
+            stdout.printf ("Error: %s\n", e.message);
+            return null;
+        }
+    }
+
+    private void create_cert_bash() {
```

Comment:
Once again the function name should not say "bash"... just call it create_cert so that when we replace the implementation the callers do not need to be changed